app/main.py

```diff
@@ -2,5 +2,15 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache = {}
+
+    def wrapper(*args):
+        if args in cache:
```

Comment:
Using function arguments as dictionary keys requires them to be hashable. If `args` contains mutable types, this will raise a `TypeError`. Consider ensuring that only hashable types are used as function arguments or convert them to a hashable type before using them as keys.